app/views/questionnaire.py

```diff
@@ -182,483 +180,103 @@ def get_hub_section(schema, questionnaire_store, section_id):
     )
 
 
-@questionnaire_blueprint.route('<block_id>/', methods=['GET'])
-@questionnaire_blueprint.route('<list_name>/<block_id>/', methods=['GET'])
+# pylint: disable=too-many-return-statements
+@questionnaire_blueprint.route('<block_id>/', methods=['GET', 'POST'])
+@questionnaire_blueprint.route('<list_name>/<block_id>/', methods=['GET', 'POST'])
 @questionnaire_blueprint.route(
-    '<list_name>/<list_item_id>/<block_id>/', methods=['GET']
+    '<list_name>/<list_item_id>/<block_id>/', methods=['GET', 'POST']
 )
 @login_required
 @with_questionnaire_store
 @with_schema
-@section_routing_path_required
-def get_block(
-    routing_path,
-    schema,
-    questionnaire_store,
-    block_id,
-    list_name=None,
-    list_item_id=None,
-):
+def get_block(schema, questionnaire_store, block_id, list_name=None, list_item_id=None):
```

Comment:
I think we should rename this from `get_block` since it now handles `POST` as well?

Reply:
`handle_block` perhaps?

Reply:
After a quick search it seems the general convention is to name the route without the get or post. I will change this from `get_block` to `block` and `get_relationship` to `relationship`.